### PR TITLE
cli for listing & bulk loading factorygirl factories

### PR DESCRIPTION
**Pull request by autotelik**
Starter for 10

**Usage**

t = bundle exec thor 

#List all available factories

t list

#Import a specified  factory

-f provide the factory to load

time taken varies depending on complexity of factory ...

simple
t wes:bulk_load:import -n 10 **-f page_applicant_contact**
```
2016-03-03 12:24:47 starting load of 10 enrollments using factory page_applicant_contact
.Bulk load of [10] Enrollments took [0.660715643] seconds
```

most complex
t wes:bulk_load:import -n 100 **-f page_complete**
```
2016-03-03 12:25:23 UTC starting load of 100 enrollments using factory page_complete
.Bulk load of [100] Enrollments took [64.845583782] seconds
```

### CODE REVIEW

**Comment by autotelik** [comment]
didn't spend any time on looking at optimisation .. could be best to just kick it off and come back in the morning .. or use it to upload a proportion then dump to sql and clone the sql changing the names, streets, postcodes etc

File: lib/tasks/bulk_load.thor [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+      puts "Factory names:\n\t#{FactoryGirl.factories.collect(&:name).sort.inspect}"
+    end
+
+    desc "import", "Generate and import Enrollments"
+
+    method_option :number, :aliases => '-n', default: 5000,
+                  :desc => "Number of Enrollments to generate"
+
+    method_option :factory, :aliases => '-f', default: :page_reviewing,
```
Comment (Cruikshanks) [inline lib/tasks/bulk_load.thor]: I would set the default to page_complete as that is more likely to be the state we want the enrolments to be in by default.

**Comment by Cruikshanks** [comment]
Based on the commit comments, is threading actually working @autotelik ? If it isn't I'd want to strip it out to avoid confusion. If it works I'm happy to leave the code related to it in.

Other than that just 1 comment to come back on.

**Comment by autotelik** [comment]
the threading code defaults to not being used so should not be confusing

 the code that uses threads is standard ruby I've used successfully before so think it's worth leaving and worth doing small test on a proper server see if it improves things, use -m <nuymber> .. if no improvement then just run without specifying anything

**Pull request merged**